### PR TITLE
Project 6

### PR DESCRIPTION
**Pull request by Javivy**
## In this milestone we:

✅Created a database based on a schema
✅Added indexes for the foreign keys of the database

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  name VARCHAR(50),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories(id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, admitted_at time, patient_id INT, CONSTRAINT patient_hist FOREIGN KEY (patient_id) REFERENCES patients(id), status VARCHAR(50));
+
+CREATE TABLE invoices (id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, total_amount decimal, generated_at time, payed_at time, medical_history_id INT, CONSTRAINT invoice_hist FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id));
+
+CREATE TABLE treatments (id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
```
Comment (Gambit142) [inline schema_based_on_diagram.sql:13]: - [ ] Based on the diagram, there is a _**many-to-many**_ relationship between the `medical_histories` table and the `treatments` table. The best way to connect these tables would be through a **_join table_**. Kindly create a join table for them and reflect the relationship in their columns. 👍

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+CREATE TABLE medical_histories(id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, admitted_at time, patient_id INT, CONSTRAINT patient_hist FOREIGN KEY (patient_id) REFERENCES patients(id), status VARCHAR(50));
+
+CREATE TABLE invoices (id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, total_amount decimal, generated_at time, payed_at time, medical_history_id INT, CONSTRAINT invoice_hist FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id));
+
+CREATE TABLE treatments (id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, 
+type VARCHAR(100), name VARCHAR(100));
+
+CREATE TABLE invoice_items (id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, unit_price DECIMAL, quantity INT, total_price DECIMAL, invoice_id INT, treatment_id INT, CONSTRAINT invoice_item FOREIGN KEY (treatment_id) REFERENCES treatments (id));
+
+ALTER TABLE invoice_items ADD CONSTRAINT fk_invoice_id FOREIGN KEY (invoice_id) REFERENCES invoices (id);
```
Comment (Gambit142) [inline schema_based_on_diagram.sql:18]: - [ ] Kindly create INDEXES for all foreign keys in this schema file. Creating INDEXES on foreign keys makes it more efficient to query tables. 👍 

#### Hint: `CREATE INDEX ON medical_histories (patient_id)`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Javivy @Oklukeok 👋, 

Good job so far!

## Highlights
- Created most tables according to schema diagrams✔️ 
- Created most foreign keys according to schema diagram ✔️ 
- Created primary keys for each table ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Javivy) [reply schema_based_on_diagram.sql:18]: Hello reviewer @Gambit142 The Index for `patient_id` has already been created, in the file schema.sql line 40

Comment (Gambit142) [reply schema_based_on_diagram.sql:18]: @Javivy, please transfer those codes to this file. Remember for this milestone, you are to create a new file where the codes for creating the tables, relationships and indexes will reside. Kindly do this and ask for another review. 👍

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**